### PR TITLE
add contributing guidelines

### PR DESCRIPTION
**Pull request by cwen0**
Signed-off-by: cwen0 <cwenyin0@gmail.com>

### What problem does this PR solve?
 

Add contributing guidelines and this pr base one #631 ,

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
+# Contributing to Chaos Mesh 
+
+Thanks for your help improving the project!
```
Comment (dcalvin) [inline CONTRIBUTING.md:3]:
```suggestion
Thanks for your help in improving the project! This document provides a step-by-step guide for general contributions to Chaos Mesh. If you want to develop a specific chaos type, you may also find [Development Guide](https://chaos-mesh.org/docs/development_guides/development_overview) useful.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
+# Contributing to Chaos Mesh 
+
+Thanks for your help improving the project! 
+
+## Getting Help 
+
+If you have a question about Chaos Mesh or have encountered problems using it, [raise an issue in Github](https://github.com/pingcap/chaos-mesh/issues) or join us the #sig-chaos-mesh channel in the [TiDB Community](https://pingcap.com/tidbslack) slack workspace..
```
Comment (dcalvin) [inline CONTRIBUTING.md:7]: Move to the end.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+- [Step 2: Run unit tests](#step-2-run-unit-tests)
+- [Step 3: Run e2e test](#step-3-run-e2e-test)
+- [Step 4: Start Chaos Mesh locally and do manual tests](#step-4-start-chaos-mesh-locally-and-do-manual-tests)
+- [Step 5: Commit and push your changes](#step-5-commit-and-push-your-changes)
+- [Step 6: Create a pull request](#step-6-create-a-pull-request)
+- [Step 7: Get a code review](#step-7-get-a-code-review)
+
+### Step 1: Make the change
+
+1. Fork the Chaos Mesh repo, and then clone it to your local:
```
Comment (yeya24) [inline CONTRIBUTING.md:25]: `to your local` sounds weird.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+```bash
+$ make ensure-kubebuilder # install some test dependencies
+$ make test
+```
+
+### Step 3: Run e2e test
+
+Chaos Mesh code must pass e2e test before being submitted. Before started, you must have [Docker](https://www.docker.com/get-started/) installed and running.
+
```
Comment (yeya24) [inline CONTRIBUTING.md:79]: I am wondering whether we really need to add this step. Running e2e locally is not easy and it is really slow...

For users who want to do contributions, open a pr and run e2e test via Jenkins is much more convenient

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+```bash
+$ ./hack/e2e.sh -h
+```
+
+### Step 4: Start Chaos Mesh locally and do manual tests
+
+1. Start a Kubernetes cluster locally. There are two options:
+
+    - Use [kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation) to start a Kubernetes cluster locally and and [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/) to access the cluster. If you install these manually, run `kind`: `kind create cluster`to start the cluster.
+
```
Comment (yeya24) [inline CONTRIBUTING.md:103]: redundant `and`

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+```bash
+$ export user={your github. profile name}
+$ git clone https://github.com/${user}/chaos-mesh.git
+```
+
+1. Set your cloned local to track the upstream repository:
+
+```bash
+$ cd chaos-mesh
+$ git remote add upstream https://github.com/pingcap/chaos-mesh
```
Comment (Gallardot) [inline CONTRIBUTING.md:36]: Append the following
```
git remote set-url --push upstream no_push
````

File: CONTRIBUTING.md (hunk 4 above)
Comment (Gallardot) [reply CONTRIBUTING.md:79]: > I am wondering whether we really need to add this step. Running e2e locally is not easy and it is really slow...
> 
> For users who want to do contributions, open a pr and run e2e test via Jenkins is much more convenient

Agree. I always open a pr and run e2e test via Jenkins.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
+# Contributing to Chaos Mesh
+
+Thanks for your interest in improving the project! This document provides a step-by-step guide for general contributions to Chaos Mesh.
+
+## Communications
+
+Before starting work on something major, please reach out to us via GitHub, Slack, email, etc. We will make sure no one else is already working on it and ask you to open a GitHub issue. Also, we will provide necessary guidance should you need it.
+
+Specifically, if you want to develop a specific chaos type, åyou may also find [Development Guide](https://chaos-mesh.org/docs/development_guides/development_overview) useful.
```
Comment (Gallardot) [inline CONTRIBUTING.md:9]:
```suggestion
Specifically, if you want to develop a specific chaos type, you may also find [Development Guide](https://chaos-mesh.org/docs/development_guides/development_overview) useful.
```

**Pull request merged**